src/main/java/uk/ac/ic/wlgitbridge/bridge/repo/GitProjectRepo.java

```diff
@@ -81,6 +96,105 @@ public void useExistingRepository(
         }
     }
 
+    @Override
+    public void runGC() throws IOException {
+        Preconditions.checkState(
+                repository.isPresent(),
+                "Repo is not present"
+        );
+        File dir = getProjectDir();
+        Preconditions.checkState(dir.isDirectory());
+        Log.info("[{}] Running git gc", projectName);
+        Process proc = new ProcessBuilder(
+                "git", "gc"
+        ).directory(dir).start();
+        int exitCode;
+        try {
+            exitCode = proc.waitFor();
+            Log.info("Exit: {}", exitCode);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        if (exitCode != 0) {
+            Log.warn("[{}] Git gc failed", dir.getAbsolutePath());
+            Log.warn(IOUtils.toString(
+                    proc.getInputStream(),
+                    StandardCharsets.UTF_8
+            ));
+            Log.warn(IOUtils.toString(
+                    proc.getErrorStream(),
+                    StandardCharsets.UTF_8
+            ));
+            try {
+                Thread.sleep(1000000);
```

Comment:
Oh I'm so sorry. I left this in while testing to see what would happen on an error on an EC2 micro machine and this was the easiest way to pause without a debugger. I will take it out and make sure there's nothing else like this.